doc/users/plotting/examples/colormap_normalizations_arbitrarynorm.py

```diff
@@ -0,0 +1,62 @@
+"""
+Demonstration of using the ArbitraryNorm classes for normalization.
+"""
+
+import numpy as np
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+
+x=np.linspace(0,16*np.pi,1024)
+y=np.linspace(-1,1,512)
+X,Y=np.meshgrid(x,y)
+
+data=np.zeros(X.shape)
+
+data[Y>0]=np.cos(X[Y>0])*Y[Y>0]**2
+
+for i,val in enumerate(np.arange(-1,1.1,0.2)):
+    if val<0:data[(X>(i*(50./11)))*(Y<0)]=val
+    if val>0:data[(X>(i*(50./11)))*(Y<0)]=val*2
+            
+figsize=(16,10)
+cmap=cm.gist_rainbow
+
+plt.figure(figsize=figsize)
```

Comment:
Can you try using the object oriented interface instead of the pyplot one? It's more powerful and flexible.